packages/common/src/discussions/HubDiscussions.ts

```diff
@@ -0,0 +1,166 @@
+}
+
+/**
+ * @private
+ * Update a Hub Discussion
+ * @param discussion the discussion to update
+ * @param requestOptions user request options
+ * @returns promise that resolves a IHubDiscussion
+ */
+export async function updateDiscussion(
+  discussion: IHubDiscussion,
+  requestOptions: IUserRequestOptions
+): Promise<IHubDiscussion> {
+  // verify that the slug is unique, excluding the current discussion
+  discussion.slug = await getUniqueSlug(
+    { slug: discussion.slug, existingId: discussion.id },
+    requestOptions
+  );
+  // get the backing item & data
+  const model = await getModel(discussion.id, requestOptions);
+  // create the PropertyMapper
+  const mapper = new PropertyMapper<Partial<IHubDiscussion>>(getPropertyMap());
+  // Note: Although we are fetching the model, and applying changes onto it,
+  // we are not attempting to handle "concurrent edit" conflict resolution
+  // but this is where we would apply that sort of logic
+  const modelToUpdate = mapper.objectToModel(discussion, model);
+  // update the backing item
+  const updatedModel = await updateModel(modelToUpdate, requestOptions);
+  // now map back into a discussion and return that
+  let updatedDiscussion = mapper.modelToObject(updatedModel, discussion);
+  updatedDiscussion = computeProps(model, updatedDiscussion, requestOptions);
+  // the casting is needed because modelToObject returns a `Partial<T>`
+  // where as this function returns a `T`
+  return updatedDiscussion as IHubDiscussion;
+}
+
+/**
+ * @private
+ * Remove a Hub Discussion
+ * @param id the discussion item id
+ * @param requestOptions request options
+ * @returns a promise
+ */
+export async function deleteDiscussion(
+  id: string,
+  requestOptions: IUserRequestOptions
+): Promise<void> {
+  const ro = { ...requestOptions, ...{ id } } as IUserItemOptions;
+  await removeItem(ro);
+  return;
+}
+
+/**
+ * @private
+ * Get a Discussion by id or slug
+ * @param identifier item id or slug
+ * @param requestOptions request options
+ * @returns a promise that resolves a IHubDiscussion
+ */
+export async function fetchDiscussion(
+  identifier: string,
+  requestOptions: IRequestOptions
+): Promise<IHubDiscussion> {
+  let getPrms;
+  if (isGuid(identifier)) {
+    // get item by id
+    getPrms = getItem(identifier, requestOptions);
+  } else {
+    getPrms = getItemBySlug(identifier, requestOptions);
+  }
+  return getPrms.then((item) => {
+    if (!item) return null;
+    return convertItemToDiscussion(item, requestOptions);
+  });
+}
+
+/**
+ * @private
+ * Convert a Hub Discussion Item into a Hub Discussion, fetching any additional
+ * information that may be required
+ * @param item the discussion item
+ * @param auth request options
+ * @returns a promise that resolves a IHubDiscussion
+ */
+export async function convertItemToDiscussion(
+  item: IItem,
+  requestOptions: IRequestOptions
+): Promise<IHubDiscussion> {
+  const model = await fetchModelFromItem(item, requestOptions);
```

Comment:
[6052](https://devtopia.esri.com/dc/hub/issues/6052) explores whether we want to have the item configuration live in the item data or in the discussions database. May need to revisit this fetching and mapping logic if we need to map entity props from a discussions database record vs item data.